### PR TITLE
[minor] allow for infinite staleness

### PR DESCRIPTION
**Pull request by runruh-godaddy**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -22,8 +22,8 @@ You can instantiate the cache with 3 configurable options: `maxAge`,
 `maxStaleness`, and `fsCachePath`.
 
 - `maxAge`: The duration, in milliseconds, before a cached item expires
-- `maxStaleness`: The duration, in milliseconds, in which expired cache
-items are still served
+- `maxStaleness`: (Optional) The duration, in milliseconds, in which expired cache
+items are still served, defaults to 0, meaning never serve data past expiration. Can also be set to `Infinity`, meaning always give at least a cached version.
```
Comment (jpage-godaddy) [inline README.md:26]:
```suggestion
items are still served. Defaults to 0, meaning never serve data past expiration. Can also be set to `Infinity`, meaning always give at least a cached version.
```

File: lib/multi-level.js [hunk 2]
```diff
@@ -36,7 +36,7 @@ function getValue(opts, key, defaultValue) {
  *
  * @param {Object} opts Options for the Client instance
  * @param {Number} [opts.maxAge=600000] The duration, in milliseconds, before a cached item expires (defaults to 600,000)
- * @param {Number} [opts.maxStaleness=0] The duration, in milliseconds, in which expired cache items are still served
+ * @param {Number} [opts.maxStaleness=0] The duration, in milliseconds, in which expired cache items are still served, supports Infinity
```
Comment (jpage-godaddy) [inline lib/multi-level.js:39]:
```suggestion
 * @param {Number} [opts.maxStaleness=0] The duration, in milliseconds, in which expired cache items are still served. Supports Infinity.
```

**Review by jpage-godaddy — APPROVED** [review]

File: lib/multi-level.js [hunk 3]
```diff
@@ -94,7 +94,8 @@ class MultiLevelCache {
     // cache hit
     const now = Date.now();
     if (item.expiry < now) {
-      if (item.expiry + getValue(opts, 'maxStaleness', this._maxStaleness) < now) {
+      const maxStaleness = getValue(opts, 'maxStaleness', this._maxStaleness);
+      if (maxStaleness !== Infinity && item.expiry + maxStaleness < now) {
```
Comment (msluther) [inline lib/multi-level.js:98]: Is this change necessary? 

```js
123 + Infinity < Date.now() // false
```

File: lib/multi-level.js [hunk 4]
```diff
@@ -160,9 +161,10 @@ class MultiLevelCache {
   async _forceRefresh(key, staleItem, updateFn, opts) {
     try {
       const value = await updateFn(key, staleItem && staleItem.value);
+      const maxAge = getValue(opts, 'maxAge', this._maxAge);
       const cacheItem = {
         value,
-        expiry: Date.now() + getValue(opts, 'maxAge', this._maxAge)
+        expiry: maxAge === Infinity ? maxAge : Date.now() + maxAge
```
Comment (msluther) [inline lib/multi-level.js:167]: again, is it necessary?
```js
123 + Infinity === Infinity
```

**Review by msluther — COMMENTED** [review]
Looking at it. I agree with the readme and test updates to be more clear about `Infinity`, but there might not be any actual code changes required here to support infinite staleness.

**Comment by runruh-godaddy** [comment]
> Looking at it. I agree with the readme and test updates to be more clear about `Infinity`, but there might not be any actual code changes required here to support infinite staleness.

You are correct, removed, but left the test and documentation updates. Started with a misconception about the terms age and staleness (inverted) where the math didn't work as-is

**Review by msluther — APPROVED** [review]
Thanks for the contribution and making the usage here more clear!
